packages/frontend-2/components/project/model-page/Header.vue

```diff
@@ -33,5 +49,87 @@ const props = defineProps<{
   project: ProjectModelPageHeaderProjectFragment
 }>()
 
-const model = computed(() => props.project.model)
+const mp = useMixpanel()
+const anyMutationsLoading = useMutationLoading()
+const updateModel = useUpdateModel()
+
+const descriptionState = ref('')
+const titleState = ref('')
+
+const canEdit = computed(() => canEditProject(props.project))
+
+const currentUpdate = computed((): UpdateModelInput => {
+  const newDescription = trim(descriptionState.value)
+  const newTitle = trim(titleState.value)
+
+  return {
+    projectId: props.project.id,
+    id: props.project.model.id,
+    description:
+      props.project.model.description !== newDescription ? newDescription : null,
+    name: props.project.model.name !== newTitle ? newTitle : null
+  }
+})
+
+const anythingToUpdate = computed(() => {
+  const updates = omit(currentUpdate.value, ['id'])
+  return Object.values(updates).some((u) => !isNullOrUndefined(u))
+})
+
+const resetTitle = () => (titleState.value = props.project.model.name || '')
+const resetDescription = () =>
+  (descriptionState.value = props.project.model.description || '')
+const resetInputs = () => {
+  resetTitle()
+  resetDescription()
+}
+
+watch(
+  () => props.project.model.description,
+  () => {
+    resetDescription()
+  },
+  { immediate: true }
+)
+
+watch(
+  () => props.project.model.name,
+  () => {
+    resetTitle()
+  },
+  { immediate: true }
+)
+
+const save = async () => {
+  if (!anythingToUpdate.value || anyMutationsLoading.value) return
```

Comment:
I see a duplicate save request when changing the title in the model page, not in the project page tho. So I suggest putting a few breakpoints there to see why it gets triggered twice with the same value, even with all of these new duplication checks.